### PR TITLE
add basic reporting functionality

### PR DESCRIPTION
**Pull request by langkilde**
add more utilities
reorganize imports

@MichelEdkrantz ready for review :)

### CODE REVIEW

File: annotell-core/annotell/core/kpi/kpi.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import time
+import json
+
+
+def _human_readable(posix_time):
+    return datetime.utcfromtimestamp(posix_time).strftime('%Y-%m-%dT%H:%M:%SZ')
+
+
+class Kpi:
+    def __init__(self, kpiType, kpiId, name):
```
Comment (MichelEdkrantz) [inline annotell-core/annotell/core/kpi/kpi.py]: Be consistent, use snake_case variables everywhere

File: annotell-core/annotell/core/kpi/kpi.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                      "Threshold:\t {}\n"
+                      "Created:\t {}\n".format(self.kpiId,
+                                               kpiResult['result'],
+                                               kpiResult['threshold'],
+                                               _human_readable(kpiResult['created'])))
+            else:
+                raise NotImplementedError('Printing results for this KPI type is not implemented yet')
+
+    def toJSON(self):
+        return json.dumps(self, default=lambda o: o.__dict__, sort_keys=True, indent=4)
```
Comment (MichelEdkrantz) [inline annotell-core/annotell/core/kpi/kpi.py]: i see this in 3 places, consider refactoring into a helper function.

File: annotell-core/annotell/core/kpi/kpi.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                raise ValueError("Invalid result keys: {}, expected: {}".format(keys, expected_keys))
+            else:
+                for item in result['items']:
+                    expected_keys = ['bin', 'value', 'count']
+                    item_keys = item.keys()
+                    if not util.valid_result_keys(item_keys, expected_keys):
+                        raise ValueError("Invalid result keys: {}, expected: {}".format(item_keys, expected_keys))
+                self.results.append(result)
+
+    def printKpiResults(self):
```
Comment (MichelEdkrantz) [inline annotell-core/annotell/core/kpi/kpi.py]: use snake case

File: annotell-core/annotell/core/kpi/kpi.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                      "Result:\t\t {}\n"
+                      "Threshold:\t {}\n"
+                      "Created:\t {}\n".format(self.kpiId,
+                                               kpiResult['result'],
+                                               kpiResult['threshold'],
+                                               _human_readable(kpiResult['created'])))
+            else:
+                raise NotImplementedError('Printing results for this KPI type is not implemented yet')
+
+    def toJSON(self):
```
Comment (MichelEdkrantz) [inline annotell-core/annotell/core/kpi/kpi.py]: use snake case

File: annotell-core/annotell/core/kpi/util.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
+from pyspark import SparkContext, SQLContext
+from pyspark.ml.feature import Bucketizer
+
+import pyspark.sql.functions as psf
+import collections
+import numpy as np
+
+KPI_TYPES = ['fraction', 'histogram']
```
Comment (MichelEdkrantz) [inline annotell-core/annotell/core/kpi/util.py]: Have you seen 
https://docs.python.org/3.7/library/enum.html
?
Might be overkill

File: annotell-core/annotell/core/kpi/kpi.py (hunk 1 above)
Comment (langkilde) [reply annotell-core/annotell/core/kpi/kpi.py]: :+1
fixed

File: annotell-core/annotell/core/kpi/kpi.py (hunk 2 above)
Comment (langkilde) [reply annotell-core/annotell/core/kpi/kpi.py]: this is overriding the built-in method, which is why it has to be everywhere afai understand. Not sure if this is correct...?

File: annotell-core/annotell/core/kpi/kpi.py (hunk 3 above)
Comment (langkilde) [reply annotell-core/annotell/core/kpi/kpi.py]: fixed 👍

File: annotell-core/annotell/core/kpi/kpi.py (hunk 4 above)
Comment (langkilde) [reply annotell-core/annotell/core/kpi/kpi.py]: hm I might have misunderstood something about this 😅 maybe the overriding idea I have is wrong

File: annotell-core/annotell/core/kpi/util.py (hunk 5 above)
Comment (langkilde) [reply annotell-core/annotell/core/kpi/util.py]: I'll stick to the simple solution for now

**Comment by langkilde** [comment]
Thanks @MichelEdkrantz :) I'll learn to do better PRs as this gets more stable.

**Pull request merged**